tests/integration/conftest.py

```diff
@@ -136,3 +138,44 @@ def spec_conditionalized_version(tmp_path):
     specfile_path = tmp_path / SPECFILE
     shutil.copyfile(SPEC_CONDITIONALIZED_VERSION / SPECFILE, specfile_path)
     return specfile_path
+
+
+@pytest.fixture(
+    params=["path", "file", "raw_string", "stringio", "bytesio", "bufferedrandom"]
```

Comment:
Why not use `text_file` and `binary_file` rather than (now ambiguous) `file` and type name? Also, I would not make the type names lowercase. Or you could do the same as with `file` and use `text_io` and `binary_io`.